library/src/main/java/com/karumi/headerrecyclerview/HeaderRecyclerViewAdapter.java

```diff
@@ -46,7 +46,7 @@
    * Invokes onCreateHeaderViewHolder, onCreateItemViewHolder or onCreateFooterViewHolder methods
    * based on the view type param.
    */
-  @Override public VH onCreateViewHolder(ViewGroup parent, int viewType) {
+  @Override public final VH onCreateViewHolder(ViewGroup parent, int viewType) {
```

Comment:
This method should not be declared as final. Doing this we are restricting the component usage based on inheritance.